app/lib/actions/remote_execution/run_hosts_job.rb

```diff
@@ -71,6 +77,16 @@ def determine_proxy(template_invocation, host, load_balancer)
 
         proxy
       end
+
+      def provider_job_class(provider)
+        case provider
+        when 'SSH'
+          RunSSHJob
```

Comment:
instead of `provider_type` which returns string that you would need to match here, you can use `template.provider` https://github.com/theforeman/foreman_remote_execution/blob/master/app/models/job_template.rb#L113 so that you get the provider object, that can define the method.